ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -65,23 +165,33 @@ module Json = struct
         ("bool", StringMap.empty)
     | DateTime ->
         ("time.Time", StringMap.empty)
-    | Enum (name, kv) ->
+    | Enum (name, kv) as ty ->
         let name = snake_to_camel name in
+        record_ty action ty enum_types ;
```

Comment:
The all types can be got easily. Don't add side effect here.

Reply:
These related code has been split to the following PR which will raised after this one got merged.